web/app/js/components/Sidebar.jsx

```diff
@@ -83,7 +83,14 @@ export default class Sidebar extends React.Component {
         <div className="sidebar">
 
           <div className={`sidebar-menu-header ${this.state.collapsed ? "collapsed" : ""}`}>
-            <ConduitLink to="/servicemesh"><img src={this.state.collapsed ? logo : wordLogo} /></ConduitLink>
+            <ConduitLink to="/servicemesh">
+              <img
+                src={this.state.collapsed ? logo : wordLogo}
+                onError={e => {
+                  // awful hack to deal with the fact that we don't serve assets off absolute paths
+                  e.target.src = e.target.src.replace(/(.*)(\/[a-zA-Z]*)(\/dist)(.*)/, "$1$3$4");
```

Comment:
Hmm, how about writing this as:

```
e.target.src = e.target.src.replace("/dist/", "/");
```

Reply:
That doesn't seem to do the same thing - what I want to do is to replace the path segment directly before the `/dist`, rather than the dist itself. So for example, on master, a request to `http://127.0.0.1:62043/api/v1/namespaces/conduit/services/web:http/proxy/namespaces/dist/foo.jpg` fails because the image is not located at `namespaces/dist`, but at `dist`.

```
let badQuery = "http://127.0.0.1:62043/api/v1/namespaces/conduit/services/web:http/proxy/namespaces/dist/foo.jpg"

badQuery.replace(/(.*)(\/[a-zA-Z]*)(\/dist)(.*)/, "$1$3$4")
"http://127.0.0.1:62043/api/v1/namespaces/conduit/services/web:http/proxy/dist/foo.jpg"

badQuery.replace("/dist/", "/");
"http://127.0.0.1:62043/api/v1/namespaces/conduit/services/web:http/proxy/namespaces/foo.jpg"
```

Reply:
Ahhh -- thanks for the explanation. I didn't read the substitution closely enough.